sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/BlobItem.java

```diff
@@ -15,6 +18,35 @@
 @JacksonXmlRootElement(localName = "Blob")
 @Fluent
 public final class BlobItem {
+
+
+    public BlobItem() {
+
+    }
+
+    /**
+     * Initializes a new blob item.
+     *
+     * @param blobItemInternal The internal structure from which to pull state.
+     */
+    public BlobItem(BlobItemInternal blobItemInternal) {
+        this.name = blobItemInternal.getName();
+        this.deleted = blobItemInternal.isDeleted();
+        this.snapshot = blobItemInternal.getSnapshot();
+        this.properties = blobItemInternal.getProperties();
+        this.metadata = blobItemInternal.getMetadata();
+        this.versionId = blobItemInternal.getVersionId();
+        this.isCurrentVersion = blobItemInternal.isCurrentVersion();
+        this.isPrefix = blobItemInternal.isPrefix();
+
+        this.tags = new HashMap<>();
```

Comment:
How do the tags get populated in swagger? Is it similar to metadata? Or completely different. Why do we need to do this set thing?

Reply:
Metadata looks like this:
<Metadata>     
        <Name>value</Name>  
</Metadata>  

Tags looks like this:
<Tags> 
    <TagSet> 
        <Tag> 
            <Key>key1</Key> 
            <Value>val1</Value> 
        </Tag> 
        <Tag> 
            <Key>key2</Key> 
            <Value>val2</Value> 
        </Tag> 
    </TagSet> 
</Tags> 
 So unfortunately there's a lot more hierarchy in the tags results, which gets generates as this gross object model that you need to call like Tags.getTagSet.getTag(i).getValue as opposed to being able to index into a map.